### DISCUSSION TITLE
Now playing - in the open - "TIDAL SDK for Web"

### DISCUSSION THREAD

**Discussion by stefanmoro** [tidal-music/discussions#40] [Announcements]
Today we're making our Web SDK publicly available, both as pre-built [artifacts](https://www.npmjs.com/org/tidal-music) and as [open source](https://github.com/tidal-music/tidal-sdk-web).

With this rollout, we introduce the possibility for web apps built on [TIDAL Developer Platform](https://developer.tidal.com/) to do playback of TIDAL preview tracks. We also invite developers to look at, contribute to, and get inspiration from some of the fundamental parts of our [web app](https://listen.tidal.com/). (Yes, our own apps built in-house rely on the very same SDK that we are now opening up.)

Check out the [blog post](https://developer.tidal.com/blog/tidal-sdk-for-web), [documentation](https://developer.tidal.com/documentation/sdk/sdk-overview), [repo](https://github.com/tidal-music/tidal-sdk-web) and [reference](https://tidal-music.github.io/tidal-sdk-web) for more information.

Enjoy building!

**Comment by BitesizedLion** [comment]
Okay that's great but when are you going to actually open up the API for us so this SDK becomes useful to us?

We want user scope and to be able to interact with the user's account, you complain about "unauthorized usage" whenever someone mentions an alternative method of doing what they want but you don't give us proper access.

We've had zero actual information regarding that. We want to make cool things with your APIs if you let us, it'd also make TIDAL a better experience for everyone.

I'd even be fine with just the ability to read a user's account like playback (what track, is it paused or not, how far along, etc.)

**Reply by stefanmoro** [reply]
Hey @BitesizedLion,

Glad to hear that you want to build what we open up.

The approach we are taking is that we open up new parts of our platform as they become ready for opening up. This means that you will gradually see more functionality, and source code, becoming available. This roll out opened up new functionality in terms of playback of preview tracks for web apps, and introduced the first piece of code of our in-house developed apps as open source.

We are still in early stages of TIDAL Developer Platform, and throughout 2024 we will see much more functionality, code and more being opened up. Things we are currently focusing on, and that we target to roll out throughout this spring are corresponding SDKs for iOS/Android and allowing TIDAL users to login through 3rd party apps built on TIDAL Developer Platform (prerequisite for many of the asks in the forum, like accessing user playlists, collection etc).

So, we are working on it.

**Reply by damg22** [reply]
Thanks for the transparency and involvement, Stefan, currently scoping out a project involving these much-needed features.

**Reply by lyzkov** [reply]
Hi, Stefan. You're doing a great job!

When do you open access for full-length track playback?

**Reply by stefanmoro** [reply]
Hi @lyzkov ,

We currently don't have a fixed date for full-length playback, but we are aiming to enable it later this year as we open up our production mode request flow.

**Reply by lyzkov** [reply]
This year! That sounds great...

**Comment by AVM-Anwar** [comment]
While I appreciate TIDAL's efforts to 'very slowly' opening up their SDK and API, since TIDAL already have very good tablet apps for Android and IOS, instead of telling us how to develop another web app similar to TIDAL's own app, why not tell us how to develop TIDAL Connect on our devices?  That will give much better end-user experience while not having to duplicate what TIDAL tablet apps already capable of doing.

**Reply by stefanmoro** [reply]
Hey @AVM-Anwar,

Please see https://github.com/orgs/tidal-music/discussions/9#discussioncomment-7307171, for current status for TIDAL Connect.

**Comment by SubtleCo** [comment]
@stefanmoro Hi Stefan! Thanks for the update. Do you know if there are any plans to query user specific data (under proper auth of course)? Specifically I'm hoping for a "currently playing" route 🤞🏻 for a project I'm working on

**Reply by aliaksei-taliuk** [reply]
Hey @SubtleCo !
Please see https://github.com/orgs/tidal-music/discussions/6 for more information about user data APIs.
Can't give any specific promises but first user data endpoints should become available pretty soon (under proper auth). 
I'm not sure about "currently playing" endpoint though, but we can try to scope it and start the initial discussions.
Stay tuned!
